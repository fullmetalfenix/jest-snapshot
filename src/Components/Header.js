import React from 'react';

const navStyles = {
  background: 'red',
  padding: '20px',
  display: 'flex',
  justifyContent: 'space-evenly',
  alignContent: 'space-around',
      }    
  const linkStyles = {
      color: 'black',
      textDecoration: "none",
  }

class Header extends React.Component {

    render() {
      
      return (
        <nav style={navStyles}>
          <a style={linkStyles} href="/">Home</a> |
          <a style={linkStyles} href="/about">About</a> |
          <a style={linkStyles} href="/other-projects">Other Projects</a> |
          <a style={linkStyles} href="/contact">Contact</a>
        </nav>
      );
    }
  }

export default Header;