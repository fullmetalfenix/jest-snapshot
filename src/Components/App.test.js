import React from "react";
import renderer from "react-test-renderer";

import Items from "./Items";
import Header from "./Header";

//Demo tests (scotch.io)

it("renders correctly when there are no items", () => {
  const tree = renderer.create(<Items />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is one item", () => {
  const items = ["one"];
  const tree = renderer.create(<Items items={items} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there are multiple items", () => {
  const items = ["one", "two", "three"];
  const tree = renderer.create(<Items items={items} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// My Tests

test("Rendering the header", () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Rendering the header inline", () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchInlineSnapshot(`
    <nav
      style={
        Object {
          "alignContent": "space-around",
          "background": "red",
          "display": "flex",
          "justifyContent": "space-evenly",
          "padding": "20px",
        }
      }
    >
      <a
        href="/"
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        Home
      </a>
       |
      <a
        href="/about"
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        About
      </a>
       |
      <a
        href="/other-projects"
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        Other Projects
      </a>
       |
      <a
        href="/contact"
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        Contact
      </a>
    </nav>
  `);
});
