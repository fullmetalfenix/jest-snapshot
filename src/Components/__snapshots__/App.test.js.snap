// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the header 1`] = `
<nav
  style={
    Object {
      "alignContent": "space-around",
      "background": "red",
      "display": "flex",
      "justifyContent": "space-evenly",
      "padding": "20px",
    }
  }
>
  <a
    href="/"
    style={
      Object {
        "color": "black",
        "textDecoration": "none",
      }
    }
  >
    Home
  </a>
   |
  <a
    href="/about"
    style={
      Object {
        "color": "black",
        "textDecoration": "none",
      }
    }
  >
    About
  </a>
   |
  <a
    href="/other-projects"
    style={
      Object {
        "color": "black",
        "textDecoration": "none",
      }
    }
  >
    Other Projects
  </a>
   |
  <a
    href="/contact"
    style={
      Object {
        "color": "black",
        "textDecoration": "none",
      }
    }
  >
    Contact
  </a>
</nav>
`;

exports[`renders correctly when there are multiple items 1`] = `
<ul>
  <li
    className="item-message"
  >
    one
  </li>
  <li
    className="item-message"
  >
    two
  </li>
  <li
    className="item-message"
  >
    three
  </li>
</ul>
`;

exports[`renders correctly when there are no items 1`] = `
<span
  className="empty-message"
>
  No items in list
</span>
`;

exports[`renders correctly when there is one item 1`] = `
<span
  className="item-message"
>
  one
</span>
`;
